### ISSUE TITLE
Forecast temperature min for today becomes null

### ISSUE THREAD

**Issue by jmot205** [bremor/bureau_of_meteorology#196]
I've noticed that partway through the day, the minimum temperature for the forecast for the current day becomes null. The max/min temperature for the day in the top right corner of the forecast card then becomes max/max.

Taken in the morning:
![Screenshot_20231031_073155_Home Assistant](https://github.com/bremor/bureau_of_meteorology/assets/44866741/cbc2dd98-a349-48cb-b0b0-22b7eefc97ff)

Taken in the evening:
![Screenshot_20231030_211725_Home Assistant](https://github.com/bremor/bureau_of_meteorology/assets/44866741/3ff6f339-2677-444a-9f54-aa8cca1e1828)

**Comment by DavidFW1960** [comment]
Yep. BOM stops publishing it in the afternoon/early evening.

**Comment by jmot205** [comment]
Ah okay, I had a suspicion it might be something like that; it's an interesting choice.

Is the desire for the HA provider to be a 1:1 mapping with what BOM publishes - i.e. having it also go null in the arvo/evening? Or is there maybe a case for caching the value when it's set in the morning and keeping it until it updates the next day so that there's always a value that can be referred to / used in automations?

My use cases are some automations in the evening based on the forecast min temperature, but being null at the time of checking means they don't work after switching to the BOM provider. I could workaround by making my own user variable that reads the forecast min in the morning and uses that value until the next day, but it feels overly clunky and won't show in the forecast card. I'd also have thought that having the forecast value remain valid until it was updated would make more sense given that at the time it clears, the forecast time is still in the future?

**Comment by DavidFW1960** [comment]
So make a template sensor that sets to the minimum and use that. There are weather cards that are much more configurable than the stock cards. Today's minimum is the forecast for 4am so not real useful. You can use tomorrows forecast minimum for an automation. Forecasts are all voodoo anyway. And no the forecast minimum time is in the PAST not the future

**Comment by jmot205** [comment]
Yeah, the template sensor was another viable option that felt overly clunky to resolve it. But it sounds like it's a moot point - I did not know that that's how the min forecast worked (4am in the morning of, instead of predicted pre-midnight the evening of). That changes my use case (and makes BOM's clearing of the value in the afternoon make much more sense to me), and I can simply switch to the forecast[1] min value - thanks for the clarification.

They may well be voodoo, but I've found BOM to have decently accurate voodoo for my location and purposes :) 

Cheers - will close this as it was simply a misunderstanding on my part.
